### PR TITLE
ci: use composite checkout action

### PR DESCRIPTION
**Pull request by karhama**
Use separate action for checking out the source code. Composite action can be found from here: https://github.com/NordicBuilder/action-checkout-west-update/

### CODE REVIEW

File: .github/workflows/license-reusable.yml [hunk 1]
```diff
@@ -70,32 +59,24 @@ jobs:
...
       working-directory: ncs
       env:
         PR_REF: ${{ github.event.pull_request.head.sha }}
       run: |
         export PATH="$HOME/.local/bin:$PATH"
         export PATH="$HOME/bin:$PATH"
-        if [ "${{ inputs.path }}" = "nrf" ]; then
-            west init -l nrf
-        else
-            west init -m ${{ inputs.nrf_repo }} --mr ${{ inputs.nrf_rev }}
```
Comment (DatGizmo) [inline .github/workflows/license-reusable.yml:96]: It looks like the new action does not support this use case.

Comment (karhama) [reply .github/workflows/license-reusable.yml:96]: Correct. I tried to search if these inputs are actually used somewhere but didn't find hits (but I might have missed something).

File: .github/workflows/oss-history.yml [hunk 2]
```diff
@@ -6,22 +6,20 @@ jobs:
...
-        uses: actions/checkout@v3
+      - name: Checkout sources
+        uses: nordicbuilder/action-checkout-west-update@main
         with:
-          path: ncs/nrf
-          fetch-depth: 0
+          git-fetch-depth: 0
+          west-update-args: ''
 
       - name: Install extra python dependencies
```
[the remark this answers is not in the document]
Comment (thst-nordic) [reply .github/workflows/oss-history.yml:15]: pip doesn't read anything except 'nrf' which is same order

**Review by thst-nordic — APPROVED** [review]

File: .github/workflows/license-reusable.yml (hunk 1 above)
Comment (DatGizmo) [reply .github/workflows/license-reusable.yml:96]: Checked as well and couldn't find it being used.

**Review by DatGizmo — APPROVED** [review]

**Pull request merged**